runbotics/runbotics-desktop/src/action/desktop/desktop.action-handler.ts

```diff
@@ -0,0 +1,292 @@
+require('@nut-tree/template-matcher');
+import fs from 'fs';
+import path from 'path';
+import { StatelessActionHandler } from 'runbotics-sdk';
+import { RunboticsLogger } from '#logger';
+import { v4 as uuidv4 } from 'uuid';
+import { createWorker } from 'tesseract.js';
+import {
+    mouse,
+    keyboard,
+    clipboard,
+    screen,
+    straightTo,
+    centerOf,
+    sleep,
+    imageResource,
+    MouseConfig,
+    KeyboardConfig,
+    Point,
+    Button,
+    Key,
+    Region,
+    FileType,
+} from '@nut-tree/nut-js';
+import {
+    DesktopActionRequest,
+    DesktopClickActionInput, 
+    DesktopTypeActionInput,
+    DesktopCopyActionInput,
+    DesktopSelectWithCursorActionInput,
+    DesktopReadContentFromClipboardActionOutput,
+    DesktopFindScreenRegionActionInput,
+    DesktopWaitForScreenRegionActionInput,
+    DesktopTakeScreenshotActionInput,
+    DesktopTakeScreenshotActionOutput,
+    DesktopReadTextFromImageActionInput,
+    DesktopReadTextFromImageActionOutput,
+    ClickTarget,
+    Coordinate,
+    MouseButton,
+    RegionObj
+} from './types';
+
+
+export default class DesktopActionHandler extends StatelessActionHandler {
+    private readonly logger: RunboticsLogger = new RunboticsLogger(DesktopActionHandler.name);
+    private readonly system: string = process.platform;
+    private readonly mouseConfig: MouseConfig = {
+        autoDelayMs: 50,
+        mouseSpeed: 600,
+    };
+    private readonly keyboardConfig: KeyboardConfig = {
+        autoDelayMs: 50,
+    };
+    // screen config -> configure directory for imageResource
+    
+    constructor() {
+        super();
+        mouse.config = this.mouseConfig;
+        keyboard.config = this.keyboardConfig;
+    }
+
+    async click(input: DesktopClickActionInput): Promise<void> {
+        const button: Button = input.mouseButton === MouseButton.LEFT ? Button.LEFT : Button.RIGHT;
+        
+        if (input.clickTarget === ClickTarget.POINT) {
+            this.checkPoint(input.x, input.y);
+            this.moveMouseToPoint(input.x, input.y);
+        } else {
+            const region = input.region;
+            this.checkRegion(region);
+            await this.moveMouseToRegion(+region.left, +region.top, +region.width, +region.height);
+        }
+        
+        if (input.doubleClick) {
+            await mouse.doubleClick(button);
+        } else {
+            await mouse.click(button);
+        }
+    }
+
+    async type(input: DesktopTypeActionInput): Promise<void> {
+        const optionalKey = input.text.substring(4); 
+        if (input.text.startsWith('Key.') && Object.keys(Key).includes(optionalKey)) {
+            await keyboard.type(Key[optionalKey]);
+        } else {
+            await keyboard.type(input.text);
+        }
+    }
+
+    async copy(input: DesktopCopyActionInput): Promise<void> {
+        if (input?.text) {
```

Comment:
input can be undefined?

Reply:
yes, it works like that:
![image](https://github.com/runbotics/runbotics/assets/63171496/83a7a7cf-a4c8-46fe-a26f-94fde006a925)

Reply:
yup, but `input?.text` indicates that `input` is possibly undefined, not `text` inside of it. imo this question mark is redundant